### ISSUE TITLE
Logging out from authentik should invalidate sessions of proxied applications

### ISSUE THREAD

**Issue by rperpe** [goauthentik/authentik#2023]
**Is your feature request related to a problem? Please describe.**
If users logout from authentik they expect that they're logged out from proxied applications (using embedded outpost and integrated proxy provider) too - without GETting `/akprox/sign_out` on every single application. It's a security risk if users don't consider this.

**Describe the solution you'd like**
As soon as `https://authentik.tld/flows/-/default/invalidation/` is called, all sessions on proxied applications must be invalidated immediately (probably via OIDC back-channel logout?).
Same is expected if forward auth proxy provider is used (assuming there exists the same issue).

**Describe alternatives you've considered**
Are there any alternatives/workarounds (instructing users to manually logout from every single application isn't considered as a workaround)?

**Additional context**
N/A

PS: Many thanks to @BeryJu for his hard work and sharing authentik on GitHub!

**Comment by xpufx** [comment]
I do agree that it's a security issue when the user logs out of one app and calls it a day not realizing all it would take to log back in is to go to the login page again. Same for other apps they haven't even considered the fate of.

However it would also confuse a lot of users when they log out of their webmail account and now their kanban software is logged out too.

The other thing is, the applications need to play along with this as well. Not all apps allow you to change what happens when you hit that log out button. Authentik needs to know when it happens for it to be able to do something, no?

This doesn't look like as straight forward as it might seem at first. If it were to be implemented somehow, I'd like the behavior to be configurable.

**Comment by BeryJu** [comment]
I agree that this should at least be configurable, if not default. There are two reasons why it isnt currently possible:

- authentik doesnt support OAuth Backchannel Logout, which is how authentik would tell an outpost that a user logged out.

This isn't the case because relatively few applications and libraries support it, so I haven't gotten around to implementing it

- The authentik Outpost can't remove sessions in the background

Because the way sessions work with gorilla/sessions, the outpost can only delete a session in the context of an HTTP request, and can only delete the session of that specific HTTP request. I briefly looked for an alternative library but didnt find anything with the required features (ability to remove sessions without HTTP request, Redis and FS Storage)

**Comment by Ornias1993** [comment]
I see some confusion about SLO (Single Log Out)
Lets seperate a few things:
1. Proxied applications and apps using forwardAuth with multiple authentik applications defined
2. SAML/OIDC
3. LDAP

1. It should be possible to invalidate other authentik applications, when one of them is logged out. forwardAuth definately  supports this.

2. When a logout request from a SAML/OIDC application hits, we can also logout applications using proxied/forwardAuth applications. However, indeed not all SAML/OIDC applications support backchannel logout and SLO (or rather: most do not)

3. LDAP does not support SLO afaik, as it just says "YAY or NAY" and sends some userdetails, so case closed (basically)

It should be relatively easy to do SLO for applications using proxied/forwardAuth, by allowing administrators to set "logout groups" for Apps that need to be logged-out together.

For SAML/OIDC this gets weird, some users might want to logout a proxygroup together with a SAML/OIDC logout, but some might not realise other SAML/OIDC sessions are not logged-out at the same time. But I think the option to also logout a group of proxied/forwardAuth applications cán technically be offered.

**Comment by QuarkZ26** [comment]
So, am I to understand that logging out of Authentik doesn't really do anything? when I log out and go back to the app, I'm still not prompted.
It seems like a major issue to me, what if I log into some public machine? Even restarting the browser doesn't do anything.

**Comment by oomenit** [comment]
I just noticed this as well. This is quite a security concern. All my forwardAuth'ed apps are staying logged in, even though I'm explicitly logging out through Authentik.

I was under the impression that every request made to **subdomain.domain.tld** would check authentication against Authentik?

I'm coming from Authelia, and with the same forwardAuth-style authentication, Authelia does redirect me to the login page on all forwardAuth'ed apps as soon as I'm logged out of Authelia itself.

Or should I need to play around with the token validity?

**Comment by slimshizn** [comment]
Needs to be fixed. I lowered token validity to 1hr but still needs to be
changed to immediately log out. Especially if the user is removed and still
has access temporarily because of it...

**Comment by QuarkZ26** [comment]
> I just noticed this as well. This is quite a security concern. All my forwardAuth'ed apps are staying logged in, even though I'm explicitly logging out through Authentik.
> 
> I was under the impression that every request made to **subdomain.domain.tld** would check authentication against Authentik?
> 
> I'm coming from Authelia, and with the same forwardAuth-style authentication, Authelia does redirect me to the login page on all forwardAuth'ed apps as soon as I'm logged out of Authelia itself.
> 
> Or should I need to play around with the token validity?

I went back to Authelia, personally. Given the last response that was given here almost a year ago. It seems crazy to me to leave something like this without any resolution. This plus other issues made me move back.

**Comment by Ornias1993** [comment]
> > I just noticed this as well. This is quite a security concern. All my forwardAuth'ed apps are staying logged in, even though I'm explicitly logging out through Authentik.
> > I was under the impression that every request made to **subdomain.domain.tld** would check authentication against Authentik?
> > I'm coming from Authelia, and with the same forwardAuth-style authentication, Authelia does redirect me to the login page on all forwardAuth'ed apps as soon as I'm logged out of Authelia itself.
> > Or should I need to play around with the token validity?
> 
> I went back to Authelia, personally. Given the last response that was given here almost a year ago. It seems crazy to me to leave something like this without any resolution. This plus other issues made me move back.

Same here, Authelia is a slow-simply-and-steady development. But it at least does, what it does, well and trustworthy.

**Comment by mrjk** [comment]
Just here as well to bump this issue, it's pretty annoying problem and a quite big security issue from my point of view. Maybe this behavior should be manageable with a switch ?

But for now, this is really annoying.

**Comment by mrjk** [comment]
I found an ugly workaround. The idea is a simple HTML page that will delete all cookies of the domain, you just need to host that page under the domain you manage, ie: `https://logout.company.com/logout.html?d=company.com`. For the POC, it ask you before actually logoff, I guess this page could be fully parametrized, but just here to validate poc workaround. 

```logout.html
<!DOCTYPE html>
<html>
<head>
  <title>AK Logout</title>
</head>
<body>

<script>

  var query = window.location.search.substring(1);
  var url = new URLSearchParams(query);
  var domain = url.get("d");
  var logout_url = "https://accounts." + domain + "/flows/-/default/invalidation/";

  // Delete all cookies
  function clean_cookies(domain) {
    var cookies = document.cookie.split(";");
    for (var i = 0; i < cookies.length; i++) {
      var cookie = cookies[i];
      var eqPos = cookie.indexOf("=");
      var name = eqPos > -1 ? cookie.substr(0, eqPos) : cookie;
      var result = name + "=;expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/; domain=" + domain;
      console.log(result);
      document.cookie = result;
    }

  }

  // Confirm user
  if (confirm('You will be disconected of all "' + domain + '" applications.')) {
      clean_cookies(domain)
      window.location.href = logout_url;
  } else {
    console.log('User aborted logout.');
  }

</script>

<p><a href="javascript:window.location.href=window.location.href">Refresh this page</a></p>
</body>
</html>
```

As you guessed, I'm not a frontend dev :')

**Comment by github-cli** [comment]
That would "only" delete the cookies, the actual token would still be valid, meaning if someone had that, they could still access the site for as long as the token is valid :-(

I am not so sure people are even aware of this!
I have been using authentik for a year now and was unaware of it until yesterday.
Even manually "logging out" via https://authentik.fqdn.com/application/o/<appName>/end-session that you place on your website does not actually log you out (you get a confirmation screen but even if you also log out of authentik itself, the session stays valid). The flow to "invalidate" a session actually does not invalidate the sessions...
It does give the impression that it should work and you should be logged out but as Berry mentioned, this "feature" (being able to log out, aka invalidating sessions) will not be implemented as its currently not possible to do so with the architecture of the outposts.

I am still a bit baffled as this would be a basic mandatory "feature" I would just assume to work.


Does anyone have any actual workarounds to this?

**Comment by Ornias1993** [comment]
>  as its currently not possible to do so with the architecture of the outposts.

This is basically the big problem why I cannot use Authentic for any significant professional usecase.

**Comment by github-cli** [comment]
^^ jumping in circles ^^
yeehaw! seems maybe this will be added
such a relief!!

**Comment by slimshizn** [comment]
@BeryJu was the recent fix supposed to log out proxied applications? E.g Say I use authentik proxy provider in front of a website with no auth. If I add a user, the user has access to said website. 

If I remove the user, delete their session, remove the explicit consent, the user then should not have access to said website. I'm still seeing the same behavior as before, where the user is still able to access the website, even after removed, or logged out.

**Comment by wrt54gl** [comment]
> @BeryJu was the recent fix supposed to log out proxied applications? E.g Say I use authentik proxy provider in front of a website with no auth. If I add a user, the user has access to said website.
> 
> If I remove the user, delete their session, remove the explicit consent, the user then should not have access to said website. I'm still seeing the same behavior as before, where the user is still able to access the website, even after removed, or logged out.

I am having this problem as well.

**Comment by Liquidtouch** [comment]
Same here. If I log out of Authentik it appears that I have logged out but If I navigate to "app.domain.com" afterwards which is using Authentik for authentication I still have access to the site. This seems like it would be a massive security issue.

**Comment by hacustom** [comment]
Seeing the same behavior on my instance.  Got things setup behind Nginx, loving the ease of setup and how well it works.  I tested out multiple methods of invalidating the user/session/token - deleting the user, changing token longevity, logging out of Authentik but was always still authenticating.

**Comment by HipyCas** [comment]
I see that this seems fixed with PR #4605, and the only thing I could find about this in the documentation is this (in the [User login stage](https://goauthentik.io/docs/flow/stages/user_login/):

> ### Terminate other sessions
> When enabled, previous sessions of the user logging in will be revoked. This has no affect on OAuth refresh tokens.

Yet when modifying the user login stage, there is no option for updating this setting. How can I configure this?

**Comment by oblivioncth** [comment]
Some clarification on the current state of affairs with this, and how in practice to ensure access to a page is revoked once a user has been logged out would be appreciated. The issue has been closed due to the aforementioned PR, but it doesn't seem like it's currently possible to directly take advantage of the changes given that the initially reported behavior appears to remain.

I've confirmed in my own setup that logging out does not invalidate access to domains behind forward auth (per-application).

**Comment by franciscomfcmaia** [comment]
Not working, currently using 2023.4.1. 

I assume that #4605 should fix this? I see this is still an issue. Someone correct me if I'm wrong.

Apart from setting sessions to 1 hour does anyone else have any solutions?

**Comment by stignarnia** [comment]
Still not working in 2023.5.0

**Comment by github-cli** [comment]
> I see that this seems fixed with PR #4605, and the only thing I could find about this in the documentation is this (in the [User login stage](https://goauthentik.io/docs/flow/stages/user_login/):
> 
> > ### Terminate other sessions
> > When enabled, previous sessions of the user logging in will be revoked. This has no affect on OAuth refresh tokens.
> 
> Yet when modifying the user login stage, there is no option for updating this setting. How can I configure this?

I believe what you are talking about is separate from this issue. This would log you out of authentik if you login someplace else… well and in that case I guess also invalidate your session cookies from the logged out PC… I haven’t tried if that combination works but I believe you are mistaking the two features here

**Comment by stignarnia** [comment]
> In my opinion that is not how this should have worked and in the end this behavior was changed as described in the PR now after this change, step 8 would not allow you top open those applications anymore (big improvement!!!)

Yeah but I tried it just now in Chrome and 2023.5.0 and didn't work.
- opened the main UI
- opened application 1
- log out from the UI
- close Chrome
- open Chrome
- navigate to application 1
- still working

I don't know if it's just my configuration that doesn't support the change for some reason, as I'm using the integrated reverse proxy

**Comment by github-cli** [comment]
I am using swag (nginx reverse proxy with authentik configs already included) and the steps you describe work for me, I dont need to close and open chrome again even, after logging out of the authentik UI, I cannot open application 1 anymore.
I run all this in a containerized environment (docker), I use the integrated outpost as I only use this for my homelab environment for stuff thats accessed mostly by only me and my wife.

But yea, I guess if Berry reopened this, there is probably a known issue...

**Comment by tograss** [comment]
I would also like too raise support and awareness for this issue. Supporting SingleLogOut is critical for most businesses, even if not all SAML/OIDC applications support this, those which do should be fully supported by authentic. I.e it works with zoom and simpsamlphp.

**Comment by masterit-dev** [comment]
Anybody found some workaround?

**Comment by scraejtp** [comment]
Same issue here, using NGINX Proxy Manager setup. Sign into Authentic ->access application -> sign out of Authentic -> still have access to application.
User login stage set to terminate all other sessions and duration of only 5 minutes. This does log me out of Authentic, but still have access to proxied applications.

Rather large bug, failing the primary function.

**Comment by Ornias1993** [comment]
> I would also like too raise support and awareness for this issue. Supporting SingleLogOut is critical for most businesses, even if not all SAML/OIDC applications support this, those which do should be fully supported by authentic. I.e it works with zoom and simpsamlphp.

Agreed, singelogout is actually the more important feature of SSO for enterprise use. It's the actual thing that gives the security in SSO setups.

**Comment by miguelangel-nubla** [comment]
Let me get this straight, login is global, logout is not.
Why is this not labeled as a high priority bug?

I personally would expect a big red alert in the documentation homepage for something like this @BeryJu

**Comment by slimshizn** [comment]
Still seeing this issue since the closure. Not fixed.

**Comment by technodrome** [comment]
A SSO system which does not log you out globally, having a high-priority bug open for **two years** without anyone clearly stating when the fix will land? Sure, I'll use it...

**Comment by skiwheelr** [comment]
I wish there was a solution for this. All i can do during testing is use incognito windows.
Is there something in the authentik script section to use or can i put some html to invalidate cache when /logout uri is hit?

**Comment by chris13524** [comment]
JWT is the fundamentally wrong approach to this it seems: https://evertpot.com/jwt-is-a-bad-default/

**Comment by HilkopterBob** [comment]
We want to implement authentik in our Enterprie-Environment, but without global, centralized logouts, this simply isnt possible.

**Comment by HilkopterBob** [comment]
We temporarily resolved the issue with a logout button that manually triggers the "outpost-logout" call. Centralized user logouts are still a highly desired feature, but we already have a test deployment of our web app that enforces user redirection centrally.

To the Authentik team,

I wanted to express my gratitude for your open ears, assistance, and code work. Your support has been invaluable, and I truly appreciate your efforts. Thank you for your help!

Warm regards,
Nick

**Comment by paulcsiki** [comment]
I can confirm that the outpost logout url work as expected however it would be nice if said url would have been visible somewhere in the Authentik Admin UI.

**Comment by stignarnia** [comment]
Where is this new button / URL? I think I didn't find it because the normal logout button still only logs out from the Authentik UI and not proxied services, at least in my testing but maybe I did something wrong

**Comment by whg517** [comment]
I am looking for a unified way to log out of sessions based on the oidc approach. When I initiate logout on the front end, I still need to choose whether to log out of the authentik account in authentik.

**Comment by katzecz** [comment]
Version 2023.10 added this feature to auto terminate all sessions when logging out via user interface logout button, but since 2023.10.3 it´s not working anymore. However, outpost logout URL does. Or am I just missing something?

**Comment by technodrome** [comment]
"Most of the comments in this thread are not helpful, especially since competing solutions like keycloak + oauth2_proxy behave the same way."

Looks like their communication and solution is not helpful, either.

**Comment by ksaadDE** [comment]
> "Most of the comments in this thread are not helpful, especially since **competing solutions like keycloak + oauth2_proxy behave the same way**."

Keycloak terminates the session  when you use the ODIC logout url, even from all Apps. So I am not sure if you are referring to a technical detail or you mean logging out from all apps (or individual Apps) in general.

**Comment by ksaadDE** [comment]
Opened a discussion on this https://github.com/goauthentik/authentik/discussions/9737

**Comment by LokeYourC3PH** [comment]
Up until finding this post, I thought I had misconfigured something, but seems like I didn't. I also would like to solve this problem, especially with NextCloud where if user1 logs in, then goes to NextCloud, calls it a day and closes the NextCloud tab but logs out of Authentik, then user2 logs in on Authentik and also clicks on NextCloud, he apparently gets logged into user1's NextCloud account. Has anyone found a fix for this yet?

**Comment by rouke-broersma** [comment]
> Up until finding this post, I thought I had misconfigured something, but seems like I didn't. I also would like to solve this problem, especially with NextCloud where if user1 logs in, then goes to NextCloud, calls it a day and closes the NextCloud tab but logs out of Authentik, then user2 logs in on Authentik and also clicks on NextCloud, he apparently gets logged into user1's NextCloud account. Has anyone found a fix for this yet?

That... Sounds like a significant bug in nextcloud.. That has nothing to do with this..

**Comment by Ma27** [comment]
> That... Sounds like a significant bug in nextcloud

How is Nextcloud supposed to know that you logged out of Authentik without being told to do so (which is basically this feature request)?

**Comment by LokeYourC3PH** [comment]
> > That... Sounds like a significant bug in nextcloud
> 
> How is Nextcloud supposed to know that you logged out of Authentik without being told to do so (which is basically this feature request)?

Exactly, thank you. A workaround I considered was running a dedicated logout flow before the upcoming authorization flow, but I am too much of a noob with Authentik yet to set that up properly. If anyone could advise, I'd much appreciate it 🙏

**Comment by rouke-broersma** [comment]
> > That... Sounds like a significant bug in nextcloud
> 
> How is Nextcloud supposed to know that you logged out of Authentik without being told to do so (which is basically this feature request)?

> then user2 logs in on Authentik and also clicks on NextCloud, he apparently gets logged into user1's NextCloud account.

Unless they are talking about a shared device for some reason, logging in to another users account on nextcloud has nothing to do with authentik.

**Comment by LokeYourC3PH** [comment]
> > > That... Sounds like a significant bug in nextcloud
> > 
> > 
> > How is Nextcloud supposed to know that you logged out of Authentik without being told to do so (which is basically this feature request)?
> 
> > then user2 logs in on Authentik and also clicks on NextCloud, he apparently gets logged into user1's NextCloud account.
> 
> Unless they are talking about a shared device for some reason, logging in to another users account on nextcloud has nothing to do with authentik.

Absolutely correct, it's a shared device. In my test cases where I use my personal account, and then a test account, this does occur. If I only were to ever use my own account on my own device it wouldn't be a problem, but this is for a shared device as you astutely observed. As far as I'm concerned, the entire usecase for this feature would mostly be for that exact reason, but keep in mind that this occurs with any OIDC/OAuth2 app, not just NextCloud, and so since some other companies also use OIDC/OAuth2 for shared devices, this has to be taken care of somehow. Once again, the best way I can imagine to workaround this right now is by making a flow which invalidates currently logged in sessions, the problem is that I have no idea how I would go about doing that.

**Comment by rouke-broersma** [comment]
I don't think you can solve this with a flow in authentik because the authentik session is actually already invalidated correctly. The issue is that authentik does not notify the oidc relying party so the for example nextcloud session is not invalidated as well. For a shared device perhaps it would be simplest to just configure the device to wipe session data upon closing the browser until authentik supports back channel logout.

**Comment by LokeYourC3PH** [comment]
> I don't think you can solve this with a flow in authentik because the authentik session is actually already invalidated correctly. The issue is that authentik does not notify the oidc relying party so the for example nextcloud session is not invalidated as well. For a shared device perhaps it would be simplest to just configure the device to wipe session data upon closing the browser until authentik supports back channel logout.

Not feasible sadly, as the Browser is used for more than just that. Also, people don't habitually close the Browser when done, they usually leave it open for the next person (it's an open terminal to be used).

**Comment by ksaadDE** [comment]
@LokeYourC3PH  the problem you have encountered is exactly the same issue with outline wiki. It has nothing to do with these Apps. Sure the Apps call also a logout route but then the Authentication provider takes over.

Authentik has this middleware site where you can re-login in the user1 account (as in your example) despite you are user2 because Authentik seemingly does not handle the logout properly.

As Authentik (see discussion) has two logout systems in place. One for the "logout from all" and one "logout from app". Which is really weird.

**Comment by Crashman1983** [comment]
This behaviour should be fixed! I also expect a global logout when using such a system.

**Comment by danohn** [comment]
I'm curious to know why Single Logout works fine for me when using Incognito but doesn't work when using a normal browser session.

**Comment by srasul** [comment]
I am using swag with included authentik configs. I have a `traefik/whoami` docker container secured via a forward-proxy provider. When trying to logout with the `outpost.goauthentik.io/sign_out`, I get told I have logged out, but I get a popup saying login is successful:

![Image](https://github.com/user-attachments/assets/b71ea1e5-54cd-4a0a-9faa-e3d4ea1b2ae6)

And then If I click on "Log back into whomai" I am do not have to do the actual login. I am forwarded to the whoami. Which is odd because I thought by going to the sign_out url, I would expect to be logged out.

**Comment by SamoLaser** [comment]
Wow. This is a glaring security issue, and this issue is still open 5 years later...

We've spent the last six weeks learning and implementing authentik, only to discover this problem _four days before the site is going to launch_. We can't release the website with such a wide open security issue. All a user has to do is log into a public computer and multiple accounts of theirs instantly become compromised. This has brought a six month project to a screeching halt right before release.

If I was under different management then I would have gotten fired over suggesting Authentik.

**Comment by LokeYourC3PH** [comment]
> Wow. This is a glaring security issue, and this issue is still open 5 years later...
> 
> We've spent the last six weeks learning and implementing authentik, only to discover this problem _four days before the site is going to launch_. We can't release the website with such a wide open security issue. All a user has to do is log into a public computer and multiple accounts of theirs instantly become compromised. This has brought a six month project to a screeching halt right before release.
> 
> If I was under different management then I would have gotten fired over suggesting Authentik.

Indeed, and I have no idea how this went unnoticed or not cared for, for so long as it has. Now mind you, this only happens with OAuth2/OIDC applications. SAML and the others for all I know (I haven't tested and needed anything but OAuth2/OIDC and SAML, so I cannot confirm this, but knowing that the others function in similar ways to SAML, I'd assume that sessions also become invalidated as well), are safe, as they get invalidated after my tests. If you have to use OAuth2/OIDC like I do for some things, you can basically expect to faceplant into this massive security issue full speed.

There is a fix that I had come up with myself, considering that OAuth2/OIDC applications basically end up being exposed to this massive security loophole, but it's not a pretty one and while it fixes the initial problem with OAuth2/OIDC sessions not being invalidated upon logout from Authentik, it introduces a new bug, although a somewhat far more minor problem that doesn't lead to outright security problems.

For those wanting to know how I fixed this, it isn't as difficult as you'd think, just very cumbersome and sad that we have to waste our time on fixes like this, when this is such a glaringly terrible oversight. Anyhow:

What you do is, you modify your (or create a new one/copy of your current) logout/invalidation flow that Authentik calls upon when you click the "Logout" button on the Authentik dashboard page or wherever else you've created a button that logs you out of the entire global Authentik session. First, you add the original logout/invalidation stage to it as stage 0. It should look like this:

![Image](https://github.com/user-attachments/assets/dfcddaf2-b8a5-41cd-8d46-6752daa90379)

Next, you add another stage to it that calls upon the application in question (for me, NextCloud) its OAuth2/OIDC logout URL. In my case for NextCloud, it's `https://xxx.xxx/apps/user_oidc/sls?` which then logs out the current user from their session. Make it stage 1. It should look like this:

![Image](https://github.com/user-attachments/assets/b538fad5-daac-44a1-a730-d6f37095ca4b)

Last, you add another stage that yet again redirects back to your Authentik instance itself as the last step. This should be stage 2. It should look as such:

![Image](https://github.com/user-attachments/assets/a1b11402-bef5-4419-af8a-e672d26e8145)

Now, if you have more OAuth2/OIDC applications, you can simply add more stages between 0 and 2 that call upon the static logout URLs of the application in question (you might need to fish for it to find it depending on which application it is).

Tada, you fixed the security loophole, but now made it so that when a user then logs in again on the same tab without it refreshing it/going to the original Authentik instance URL, they will automatically be redirected to whichever application they last used upon login, as opposed to landing on the Authentik dashboard/instance page. This is a rather small annoyance that I hope to fix, maybe someone else here can help me with that now that I've given y'all pointers on how to fix such a ludicrous and actually kind of outrageous exploit.

Peace ✌️

**Comment by SamoLaser** [comment]
@LokeYourC3PH 

This isn't really a solution or reliable workaround. What happens in your Nextcloud app is that Nextcloud specifically will take a GET request to that URL, transform it into a POST request, and add a token_hint in the headers in order to log the user out. This doesn't work with all apps, as such a feature would have to be intentionally developed into the application itself. I think you just got lucky with your Nextcloud app. For many webapps, the user clicks "logout" within the app as a POST request with all the information needed to log them out forwarded in the headers.

The issue I'm having is that the end-sessions don't work with OIDC. The proxy logouts that this issue initially addressed work properly after the update, but not OIDC.

For reference, you don't need to apply redirects to the global invalidation flow. The proper way to do it is to bind expression policies to the logout stage. You make sure these policies are executed before the stage (ie. when the user clicks "sign out of authentik" the policies get executed right away). Then the logout stage completes. Then you optionally redirect them to a page that states they've logged out successfully. 
Example:
`context['flow_plan'].context['redirect'] = 'https://example.com/application/o/yourapp/end-session/'
return True`
This is what the application/o/yourapp/end-session URI is supposed to do. 

There's a separate security issue with OIDC in general that others reported. When logging out of an OIDC application, the token still remains valid and the user can just click "login with OIDC" again and they don't need to enter credentials. The workaround for this has been to set a shorter token validity period. 

But if you set up all webapps properly to function behind Authentik (ie. the login pages for your apps are effectively disabled), then this problem with OIDC is avoided because an Authentik login is required now instead of the "login with OIDC" option being available to anyone using the device.

Because Authentik doesn't actually end the session when I instruct it to (yes, the policies are all True and passing), I'm now hit with this issue and it's even worse because it sends a user to their homepage instead of a login page where a curious person would have to first click "login with OIDC" to see that homepage.

**Comment by firestrife23** [comment]
I am truly astonished that this issue persists. It seems straightforward to implement session invalidation mechanisms, such as expiring logged-in sessions. For instance, a token could be set to expire after a predefined period—say, 24 hours—after which users would be required to log in again for revalidation. If it were up to me, I would adopt a zero-trust approach to enhance security.

**Comment by SamoLaser** [comment]
> I am truly astonished that this issue persists. It seems straightforward to implement session invalidation mechanisms, such as expiring logged-in sessions. For instance, a token could be set to expire after a predefined period—say, 24 hours—after which users would be required to log in again for revalidation. If it were up to me, I would adopt a zero-trust approach to enhance security.

You can. 
Providers > Actions > Advanced Protocol Settings > Refresh Token Validity

**Comment by PrivatePuffin** [comment]
@BeryJu At this point this has so much feedback, from both home and enterprise users that this should get attention.
Its both a security and a usability issue and basically makes authentik an impossible choice for many users.

**Comment by stevelogik** [comment]
While this issue exists you essentially can't have Authentik internet front facing or access it on any shared system. This immediately makes it a no go for production. Hopefully the Authentik team fixes this soon. I agree with the post above that it makes it an impossible choice while this issue exists.

**Comment by siment** [comment]
I don't know now @rissson closed this with references to two-year old application updates.

It still is the case that logging out of Authentik does not log out of some of my forward auth reverse proxied applications. This is true both for domain level and single application level forward auth.

To be honest, I only have one problematic application, and I did experiment with OpenID Connect on it. Those configurations were removed and replaced by multiple iterations of forward auth providers.

The fact that I can still access applications even after logging out of Authentik prevents me from moving to production, and I can only repeat some of the frustrations expressed earlier in this thread. This seems like a rather critical flaw in the architecture of an application of this nature.

Still love the project, though, and I am very grateful for the efforts of all contributors.

I am using Docker tag 2025.6.0

**Comment by rissson** [comment]
@siment I'm not sure I am following your post.

If your issue is with the proxy provider still allowing requests to go through after you've revoked the session (i.e. logged out of authentik), then that is a security issue. Please follow our [security policy](https://docs.goauthentik.io/docs/security/policy#reporting-a-vulnerability) to report this.

If your issue is with the OIDC implementation in authentik not supporting backchannel logout, as mentioned, the discussion is happening at #12710, which we still plan on doing relatively soon.

Otherwise, please open a new issue.

**Comment by siment** [comment]
Holy moly, I am impressed by your quick response. I will submit a report as you suggest.

Thanks for the tip regarding https://github.com/goauthentik/authentik/issues/12710, I will follow it with interest.

**Comment by siment** [comment]
Update:

I am closer to figuring out what caused this. And it does happen on all applications, not just the one I mentioned in the previous post.

Whether it is a security bug is debatable, but it something to consider at least.

The problem has to do with multiple sessions on the same user being created in Authentik after restart of Authentik containers. I had a couple dozen sessions - all in my name - and they were probably interfering with application sessions.

Deleting sessions for my user in Authentik admin interface seemed to to the trick. I do however notice that duplicate sessions are created after restart.

Would it make sense to delete all sessions on a restart - or maybe implement an optional feature for this?

All the best,
Simen

**Comment by EsEnZeT** [comment]
> > I am truly astonished that this issue persists. It seems straightforward to implement session invalidation mechanisms, such as expiring logged-in sessions. For instance, a token could be set to expire after a predefined period—say, 24 hours—after which users would be required to log in again for revalidation. If it were up to me, I would adopt a zero-trust approach to enhance security.
> 
> You can. Providers > Actions > Advanced Protocol Settings > Refresh Token Validity



> > I am truly astonished that this issue persists. It seems straightforward to implement session invalidation mechanisms, such as expiring logged-in sessions. For instance, a token could be set to expire after a predefined period—say, 24 hours—after which users would be required to log in again for revalidation. If it were up to me, I would adopt a zero-trust approach to enhance security.
> 
> You can. Providers > Actions > Advanced Protocol Settings > Refresh Token Validity

I don't see such option for proxy/forward auth

<img width="767" height="549" alt="Image" src="https://github.com/user-attachments/assets/d68a0731-08e8-4852-ba04-e3383eea6476" />

**Comment by ahmedraza118** [comment]
I am facing an issue logging out from Idp when while i am logged out from my frontend app and main OIDC, but from the idp i am unable to clear session, let me explain: 
I am using two instances of authentik 
1. vm2 is the main OIDC that is connected to my frontend app
2. vm3 is the idp used for federated authentication now i am logged in through vm3 to vm2 and eventually logged in my frontend app, but when i logout it clear session of my frontend app and vm2 but it is unable to chain to vm3 the idp session clear like it should in single signout and after all sessions clear redirect back to my frontend app? any solution for that?
